### PR TITLE
feat: keep drawer state

### LINKED ISSUE #277: Keep drawer state
When switching between tabs, keep track of the drawer state (`open`) and persist it.

### PR DESCRIPTION
**Pull request by hasanagh**
keeps drawer state

closes #277

### CODE REVIEW

**Review by pyphilia — APPROVED** [review]
Really nice! It won't annoy JC anymore :P

File: src/reducers/layout.js [hunk 1]
```diff
@@ -17,6 +22,8 @@ export default (state = INITIAL_STATE, { type, payload }) => {
       return state.setIn(['tools', 'open'], false);
     case SET_TOOLS_WIDTH:
       return state.setIn(['tools', 'width'], payload);
+    case CHANGE_SIDE_BAR_VIEW_STATE:
+      return state.setIn(['sideBarState'], payload);
```
Comment (juancarlosfarah) [inline src/reducers/layout.js:26]: I agree with @pyphilia. In fact, I would follow the strategy followed by tools. With a depth of two levels.

**Review by juancarlosfarah — CHANGES_REQUESTED** [review]
Looks good! Just a minor change for robustness.

**Comment by juancarlosfarah** [comment]
@pyphilia, @hasanagh, what's the status here?

File: src/actions/layout.js [hunk 2]
```diff
@@ -26,12 +31,25 @@ const setHeight = (id, height) => {
...
+const getHeight = (id) => {
   const height = localStorage.getItem(id);
   if (height) {
     return parseInt(height, RADIX);
   }
   return null;
 };
 
-export { setHeight, getHeight, openTools, closeTools, setToolsWidth };
+const changeSideBarViewState = (state) => (dispatch) =>
```
Comment (pyphilia) [inline src/actions/layout.js:42]: Maybe change this function's name to match with the type eg. `setSideBarIsOpen`

**Review by pyphilia — APPROVED** [review]
Thank you for the changes @hasanagh , just this minor change to match the rest of the code.

File: src/actions/layout.js (hunk 2 above)
Comment (juancarlosfarah) [reply src/actions/layout.js:42]: @hasanagh, so this is the only thing missing. Right @pyphilia ?

Comment (hasanagh) [reply src/actions/layout.js:42]: Yes! Will refactor and push the changes!

**Comment by pyphilia** [comment]
All good for me!

**Pull request merged**